arangod/Transaction/Methods.cpp

```diff
@@ -3007,7 +3023,7 @@ Future<Result> Methods::replicateOperations(
             operation),
         state()->id(),
         replication2::replicated_state::document::ReplicationOptions{});
-    return Result{};
+    return performIntermediateCommitIfRequired(collection->id());
```

Comment:
Why? This is happening immediately after `replicateOperation`, which itself should already call `performIntermediateCommitIfRequired`. I think we should instead just return the result of the `replicateOperation` call.

Reply:
Sorry, I was confused by the diff! All good!